replicatelogs.go

```diff
@@ -372,9 +378,45 @@ func (v *VerifiedReplica) ReplicateVerifiedUpdates(
 		if local == nil {
 			return opts
 		}
+
 		return append(opts, massifs.WithTrustedBaseState(local.MMRState))
 	}
 
+	// on demand promotion of a v0 state to a v1 state, for compatibility with the consistency check.
+	trustedBaseState := func(local *massifs.VerifiedContext) (massifs.MMRState, error) {
+
+		if local.MMRState.Version > int(massifs.MMRStateVersion0) {
+			// this will just work, until v1 seals reach production no customer can encounter this.
+			return local.MMRState, nil
+		}
+
+		if !v.cCtx.Bool("enable-v1seals") {
+			// The user has not explictly enabled the new seal format, so we return the legacy state unchanged.
+			// THis will "just work" for production. Use against a pre-release endpoint will fail verification.
+			return local.MMRState, nil
+		}
+
+		// Ok, production case. We need to promote the legacy base state to a V1
```

Comment:
you mean pre-release case?

Reply:
Sort of. I mean production case *before we release the forestrie update*. Before we release forestrie, veracity needs to deal with prod being on v0 seals. Do you want me top change the comment or does the code look wrong ?

Reply:
Yeah, I guess its all production. At this point, we expect the avid + forestrie release to be done. We have a local seal in v0 format and want to promote it to a v1 seal so it is a valid trusted base state for the remote v1 seals. I'll update the comment

Reply:
comment updated, please re-open if that still seems unclear or the code looks wrong